builtin/providers/ultradns/resource_ultradns_dirpool.go

```diff
@@ -566,10 +567,18 @@ func mapFromGeoInfos(gi *udnssdk.GeoInfo) []map[string]interface{} {
 func hashIPInfoIPs(v interface{}) int {
 	var buf bytes.Buffer
 	m := v.(map[string]interface{})
-	buf.WriteString(fmt.Sprintf("%s-", m["start"].(string)))
-	buf.WriteString(fmt.Sprintf("%s-", m["end"].(string)))
-	buf.WriteString(fmt.Sprintf("%s-", m["cidr"].(string)))
-	buf.WriteString(fmt.Sprintf("%s", m["address"].(string)))
+	if m["start"] != nil {
```

Comment:
Why do these need nil checks now?

Reply:
Was this code here without nil checks before we started changing things?

Reply:
It is supposed to get default values from the configuration reading code. Unit tests don't go through there, and it does seem unsafe to assume configs will always exist.